menas/ui/components/schema/schemaDetail.controller.js

```diff
@@ -172,8 +172,11 @@ sap.ui.define([
 
       if (status === 500) {
         const sSchemaType = this.byId("schemaFormatSelect").getSelectedItem().getText();
+        let oRawResponse = JSON.parse(oParams.getParameter("responseRaw"));
+        const errorMessage = oRawResponse.message;
```

Comment:
Will `oRawResponse` always defined and guaranteed to have `message` parameter?

Reply:
To this API call, yes.

Reply:
I dunno. Especially for status 500, to rely that the server will return an expected format of response seems little unsafe. 🤔

Reply:
It's true it shouldn't be returning `500 Internal Server Error`. We have to catch and handle the error and return `400 Bad Request`

Reply:
1. Let's create a task to change the response code.
2. On `500` let's have an additional check here that the response is valid JSON that contains a `message` field.